src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -132,14 +132,7 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
             bool hasErrors,
             DiagnosticBag diagnostics)
         {
-            SyntaxNode innerExpression = node.Expression.SkipParens();
-            if (innerExpression.Kind() == SyntaxKind.DefaultLiteralExpression)
-            {
-                diagnostics.Add(ErrorCode.ERR_DefaultPattern, innerExpression.Location);
-                hasErrors = true;
-            }
-
-            return BindConstantPattern(node, inputType, node.Expression, hasErrors, diagnostics, out _);
+            return BindConstantPattern(node, inputType, node.Expression, hasErrors, diagnostics, out _, reportDefaultMissingType: true);
```

Comment:
>reportDefaultMissingType: true [](start = 104, length = 30)

Would it make sense to keep reporting ERR_DefaultPattern for this situation for the benefit of the user?